src/PerfView/EtwEventSource.cs

```diff
@@ -327,6 +327,15 @@ public override void ForEach(Func<EventRecord, bool> callback)
                         }
                     }
 
+                    if (FilterQueryExpressionTree != null)
```

Comment:
I like that this is completely opt-in and so if a user doesn't specify a filter, then none of this runs.